### PR TITLE
fix(site): Fix template version editor rename

### PR DESCRIPTION
**Pull request by BrunoQuaresma**

### CODE REVIEW

File: site/src/components/TemplateVersionEditor/FileDialog.tsx [hunk 1]
```diff
@@ -123,14 +131,14 @@ export const RenameFileDialog: FC<{
...
       description={
-        <Stack spacing={1}>
-          <Typography>
-            Rename {`"${filename}"`} to something else. This path can contain
-            slashes too!
-          </Typography>
+        <Stack>
+          <p>
+            Rename <strong>{filename}</strong> to something else. This path can
+            contain slashes too!
```
Comment (code-asher) [inline site/src/components/TemplateVersionEditor/FileDialog.tsx:140]: I see above we changed `!` to `.`, should we do that here too?

**Review by code-asher — COMMENTED** [review]
Looking good!  I tested out renaming and deleting but when I rename the old file stays and when I delete it closes the editor but the file is still there.

tree

**Comment by BrunoQuaresma** [comment]
@code-asher good catch! We probably need to keep track of the deleted files 🤔.... @kylecarbs I think this feature is more tricky than it looks like. What do you think about we remove it for now with the additional file? So we could safely release it until we have this figured out. Some complexity that I see is a user trying to add or rename a file type that is not supported.

**Comment by BrunoQuaresma** [comment]
Hm... thinking better, maybe it is not that much work. I will try to fix it.

**Comment by BrunoQuaresma** [comment]
@code-asher could you please try it again?

**Review by kylecarbs — APPROVED** [review]
Didn't try this out, but feel free to merge whenever you think it's good!

**Review by code-asher — COMMENTED** [review]
Rename and delete are working much better now!

There are some quirks:

- Creating a file without a certain extension (for example `test` or `test.h`) always creates a folder.  Ideally I think there would be separate new folder/file buttons.
- Renaming a file without a certain extension creates a folder.  You can then do weird things like add files underneath this "folder" which is actually a file.  A file should always be a file even if renamed regardless of the extension.
- You can add a file under a file (`main.tf/test.md`) which seems to cause the file to get ordered like a folder but the new file will not actually appear.  This should probably be disallowed.
- Renaming a folder that has children causes a crash.
- Renaming a folder that has no children with a certain extension (like `test.md`) causes it to become a file.  A folder should always be a folder even if renamed regardless of the extension.
- Sorting seems to be a bit random.  If I add `test` it shows at the bottom then if I add `test/test` it shows up at the top.

**Comment by BrunoQuaresma** [comment]
I tried to solve all the Asher points but I had to disable folder creation or renaming for now. I think what we have right now is good enough but stills buggy for certain use cases. I think this is good for now and we can get back into it later and do a major refactoring and add different validation and flow for folders and files.

**Comment by code-asher** [comment]
Tested it out and it looks solid, the only thing I noticed is that I can still create/rename folders by leaving out the dot (for example `maintf` or `md`).

**Pull request merged**